schematron/Consolidation.sch

```diff
@@ -1277,8 +1277,8 @@ Schematron generated from Trifolia on 7/20/2021
       <sch:assert id="a-1098-7499" test="count(cda:templateId[@root='2.16.840.1.113883.10.20.22.4.16'][@extension='2014-06-09'])=1">SHALL contain exactly one [1..1] templateId (CONF:1098-7499) such that it SHALL contain exactly one [1..1] @root="2.16.840.1.113883.10.20.22.4.16" (CONF:1098-10504). SHALL contain exactly one [1..1] @extension="2014-06-09" (CONF:1098-32498).</sch:assert>
       <sch:assert id="a-1098-7500" test="count(cda:id) &gt; 0">SHALL contain at least one [1..*] id (CONF:1098-7500).</sch:assert>
       <sch:assert id="a-1098-7507" test="count(cda:statusCode)=1">SHALL contain exactly one [1..1] statusCode (CONF:1098-7507).</sch:assert>
-      <sch:assert id="a-1098-7508-c" test="cda:effectiveTime[@xsi:type='IVL_TS']">SHALL contain exactly one [1..1] effectiveTime (CONF:1098-7508) such that it</sch:assert>
-      <sch:assert id="a-1098-28499-c" test="not(tested-here)">**SHALL** contain exactly one [1..1] @xsi:type="PIVL_TS" or "EIVL_TS" (CONF:1098-28499).</sch:assert>
+      <sch:assert id="a-1098-7508-c" test="count(cda:effectiveTime[(cda:low or @value) and not(cda:low and @value)])=1">SHALL contain exactly one [1..1] effectiveTime (CONF:1098-7508) such that it SHALL contain either a low or a @value but not both (CONF:1098-32890).</sch:assert>
```

Comment:
There's a constraint on `@value`, so we typically wouldn't allow bypassing that constraint with a `@nullFlavor`. The proper way to send unknown medication timing would be:

```
<effectiveTime>
  <low nullFlavor="UNK" />
</effectiveTime>
```

...while  `<effectiveTime nullFlavor="UNK" />` would be invalid.

It would also make the "such that it" branching logic weird. There SHALL be exactly 1 such that it contains low or `@value`. The branching is based on low or `@value`. If you add `@nullFlavor` to that, then it's impossible to have multiple effectiveTimes with nullFlavor (not that it would really make any sense...but I suppose someone may want to say they don't know the timing AND they don't know the frequency....)